src/Pages/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,106 @@
+import { PeopleFilters } from '../../components/PeopleFilters';
+import { Loader } from '../../components/Loader';
+import { useEffect, useState } from 'react';
+import { useParams, useSearchParams } from 'react-router-dom';
+import { getPeople } from '../../api';
+import { PeopleList } from '../../components/PeopleList';
+import { Person } from '../../types';
+
+const peopleFilter = (visiblesPeople: Person[], query: string) => {
+  return visiblesPeople.filter(
+    person =>
+      person.name.toLowerCase().includes(query.trim().toLowerCase()) ||
+      person.motherName?.toLowerCase().includes(query.trim().toLowerCase()) ||
+      person.fatherName?.toLowerCase().includes(query.trim().toLowerCase()),
+  );
+};
+
+export const PeoplePage = () => {
+  const [searchParams] = useSearchParams();
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex');
+  const centuries = searchParams.getAll('centuries') || [];
+
+  const [people, setPeople] = useState<Person[]>([]);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [isLoading, setIsLoading] = useState(false);
+
+  let visiblesPeople = [...people];
+
+  if (query) {
+    visiblesPeople = peopleFilter(visiblesPeople, query);
+  }
+
+  if (sex) {
+    visiblesPeople = visiblesPeople.filter(personSex => personSex.sex === sex);
+  }
+
+  if (centuries.length > 0) {
+    visiblesPeople = visiblesPeople.filter(person =>
+      centuries.some(centery => Math.ceil(person.born / 100) === +centery),
+    );
+  }
+
+  const { personId = '' } = useParams();
+
+  useEffect(() => {
+    setIsLoading(true);
+    getPeople()
+      .then(setPeople)
+      .catch(error => {
+        setErrorMessage(`Something went wrong: ${error.message}`);
+      })
+      .finally(() => setIsLoading(false));
+  }, [errorMessage]);
+
+  const getFather = (name: string | null) =>
+    visiblesPeople.find(person => person.name === name);
+
+  const getMother = (name: string | null) =>
+    visiblesPeople.find(person => person.name === name);
```

Comment:
Better would be to prepare you people array before rendering, and to add `mother` and `father` to every object

Reply:
> Better would be to prepare you people array before rendering, and to add `mother` and `father` to every object

Not fixed